packages/pigeon/e2e_tests/test_objc/lib/dartle.dart

```diff
@@ -128,3 +128,20 @@ class Api {
     }
   }
 }
+
+abstract class MockApi {
+  SearchReply search(SearchRequest arg);
+}
+
+void MockApiSetup(MockApi api) {
```

Comment:
"mock handler" is already used terminology for the same concept elsewhere: https://api.flutter.dev/flutter/services/MethodChannel/setMockMethodCallHandler.html

Reply:
Ah I see where you're going. That thing is a mock though. It's taking a production class and swapping it out for an alternate one which records and stubs interactions. 

This doesn't replace a production class. It's used in parallel with the production class to is a separate "working" class that has a different behavior that the real one. 

It'd be like we had `client` and `server` and we want to test `client`. We can have a `mockClient` which is what we all want but since we can't do it easily for existing tests, we're using `client` with a `fakeServer`. But if we do, we can't make the `fakeServer` be named `mockClient`. That would be really confusing. 

We can call this TestHost[name of API] for instance. Since we're specifically doing the 'host' part of the class annotated by `HostApi`

Reply:
I renamed this class to `TestHostApi`.